src/mongo/crypto/crypto_openssl.cpp

```diff
@@ -45,19 +45,26 @@ namespace crypto {
  * Computes a SHA-1 hash of 'input'.
  */
 bool sha1(const unsigned char* input, const size_t inputLen, unsigned char* output) {
-    EVP_MD_CTX digestCtx;
-    EVP_MD_CTX_init(&digestCtx);
-    ON_BLOCK_EXIT(EVP_MD_CTX_cleanup, &digestCtx);
+    EVP_MD_CTX *digestCtx;
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+    digestCtx = (EVP_MD_CTX*) malloc(sizeof(EVP_MD_CTX));
+    EVP_MD_CTX_init(digestCtx);
+    ON_BLOCK_EXIT(free, digestCtx);
+    ON_BLOCK_EXIT(EVP_MD_CTX_cleanup, digestCtx);
+#else
+    digestCtx = EVP_MD_CTX_new();
```

Comment:
Please change the EVP_MD_CTX* into a unique_ptr. We're trying to move away from ON_BLOCK_EXIT, because it's a little inflexible.

We have two idioms to deal with this, MakeGuard and std::unique_ptr with a custom deleter function. I personally lean toward the latter, especially if we're defining our own copy of EVP_MD_CTX_free.

Then, the complete type of the unique_ptr is std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)> Take a look at the type alias 'UniqueSSLContext' in src/mongo/util/net/ssl_manager.cpp.